datacapture/src/main/java/com/google/android/fhir/datacapture/enablement/EnablementEvaluator.kt

```diff
@@ -86,33 +85,42 @@ internal object EnablementEvaluator {
     return when (val value = questionnaireItem.enableBehavior.value) {
       EnableWhenBehaviorCode.Value.ALL ->
         enableWhenList.all {
-          evaluateEnableWhen(questionnaireItem.type, it, questionnaireResponseItemRetriever)
+          evaluateEnableWhen(it, questionnaireItemAndQuestionnaireResponseItemRetriever)
         }
       EnableWhenBehaviorCode.Value.ANY ->
         enableWhenList.any {
-          evaluateEnableWhen(questionnaireItem.type, it, questionnaireResponseItemRetriever)
+          evaluateEnableWhen(it, questionnaireItemAndQuestionnaireResponseItemRetriever)
         }
       else -> throw IllegalStateException("Unrecognized enable when behavior $value")
     }
   }
+}
 
-  /**
-   * Returns whether the `enableWhen` constraint is satisfied.
-   *
-   * @param questionnaireResponseItemRetriever function that returns the
-   * [QuestionnaireResponse.Item] with the `linkId`, or null if there isn't one.
-   */
-  private fun evaluateEnableWhen(
-    type: Questionnaire.Item.TypeCode,
-    enableWhen: Questionnaire.Item.EnableWhen,
-    questionnaireResponseItemRetriever: (linkId: String) -> QuestionnaireResponse.Item?
-  ): Boolean {
-    val responseItem = questionnaireResponseItemRetriever(enableWhen.question.value) ?: return true
-    return if (QuestionnaireItemOperatorCode.Value.EXISTS == enableWhen.operator.value) {
-      (responseItem.answerCount > 0) == enableWhen.answer.boolean.value
-    } else {
-      responseItem.contains(enableWhenTypeToPredicate(enableWhen, type))
-    }
+/** Result class to unpack questionnaireItem and questionnaireResponseItem */
+data class Result(
```

Comment:
```suggestion
data class QuestionnaireItemWithResponse(
```

or something similar?

Reply:
Name changed to `QuestionnaireItemWithResponse`.